gtda/externals/python/ripser_interface.py

```diff
@@ -122,6 +122,28 @@ def get_greedy_perm(X, n_perm=None, metric="euclidean"):
     return idx_perm, lambdas, dperm2all
 
 
+def _resolve_symmetry_conflicts(coo):
+    _row, _col, _data = coo.row, coo.col, coo.data
+
+    upper_triangle = _col >= _row
+    row = _row[upper_triangle] 
+    col = _col[upper_triangle]
+    data = _data[upper_triangle]
+    below_diag = np.logical_not(upper_triangle)
+    _info_below_diag = _row[below_diag], _col[below_diag], _data[below_diag]
+
+    upper_triangle_indices = set(zip(row, col))
+    additions = tuple(zip(*((j, i, x) for (i, j, x) in zip(*_info_below_diag)
+                            if (j, i) not in upper_triangle_indices)))
```

Comment:
I'm not confident that in 3 months we will understand quite easily what's going on here

Reply:
Indeed. The function still needs documenting, as I wrote. I'll include inline comments.

Reply:
Added.